src/backend/distributed/commands/foreign_constraint.c

```diff
@@ -1325,60 +1309,22 @@ GetForeignConstraintCommandsToReferenceTable(ShardInterval *shardInterval)
 	/* revert back to original search_path */
 	PopOverrideSearchPath();
 
+	commandList = lappend(commandList, "SET citus.skip_constraint_validation TO OFF;");
```

Comment:
SET preserves this for the rest of the session, which is dangerous. Better to use RESET